.github/workflows/upload_new_tag_to_pypi.yml

```diff
@@ -0,0 +1,58 @@
+name: Publish Python 🐍 distributions 📦 to PyPI and TestPyPI
+
+on: push
+
+jobs:
+  build-n-publish:
+    name: Build and publish Python 🐍 distributions 📦 to PyPI and TestPyPI
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      # with:
+      #   python-version: 3.9
+
+    - name: Install pypa/build
+      run: >-
+        python -m
+        pip install
+        build
+        --user
+    - name: Build a binary wheel and a source tarball
+      run: >-
+        python -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+
+    - name: Install twine
+      run: >-
+        python -m
+        pip install
+        twine
+
+    - name: Check package metadata
+      run: >-
+        twine check dist/*
+
+    # NOTE: "Secrets" page, not "Environments"
+    # NOTE: testpypi upload with versioneer only works on new tags.
+    # - name: Publish distribution 📦 to Test PyPI
+    #   uses: pypa/gh-action-pypi-publish@master
+    #   with:
+    #     user: __token__
+    #     password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+    #     repository_url: https://test.pypi.org/legacy/
+    #     verbose: true
+
+    - name: Publish distribution 📦 to PyPI
+      if: startsWith(github.ref, 'refs/tags')

```

Comment:
This line only prevents this single action from running unless the commit is tagged.

Reply:
Correct.